### PR TITLE
Cleaned up countitem-type script commands

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #3612

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Separated the countitem-type script commands into their own functions.
  * Created a global count function to be used between them.
Thanks to @kisuka's suggestion!

### CODE REVIEW

**Comment by RadianFord** [comment]
Is this good to go?

**Comment by aleos89** [comment]
Should be, yes.

**Comment by lighta** [comment]
Will there be a depreciation notice about the old countitem for the script using them ?

**Comment by aleos89** [comment]
The commands aren't deprecated, just move into their own functions.

File: src/map/script.cpp [hunk 1]
```diff
@@ -6870,39 +6795,229 @@ BUILDIN_FUNC(countitem)
...
+		script_pushint(st, -1);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	if (script_isstring(st, 2)) // item name
+		id = itemdb_searchname(script_getstr(st, 2));
+	else // item id
+		id = itemdb_exists(script_getnum(st, 2));
+
+	if (id == NULL) {
```
Comment (Normynator) [inline src/map/script.cpp]: redundant nullptr check

File: src/map/script.cpp [hunk 2]
```diff
@@ -6870,39 +6795,229 @@ BUILDIN_FUNC(countitem)
...
+		if (!script_rid2sd(sd))
+			return SCRIPT_CMD_FAILURE;
+	}
+
+	if (script_isstring(st, 2)) // item name
+		id = itemdb_searchname(script_getstr(st, 2));
+	else // item id
+		id = itemdb_exists(script_getnum(st, 2));
+
+	if (id == NULL) {
```
Comment (Normynator) [inline src/map/script.cpp]: redundant nullptr check

File: src/map/script.cpp [hunk 3]
```diff
@@ -6870,39 +6795,229 @@ BUILDIN_FUNC(countitem)
...
+		if (!script_rid2sd(sd))
+			return SCRIPT_CMD_FAILURE;
 	}
 
+	if (script_isstring(st, 2)) // item name
+		id = itemdb_searchname(script_getstr(st, 2));
+	else // item id
+		id = itemdb_exists(script_getnum(st, 2));
+
+	if (id == NULL) {
```
Comment (Normynator) [inline src/map/script.cpp]: redundant nullptr check

**Pull request merged**